hibernate-reactive-core/src/test/java/org/hibernate/reactive/IdentityGeneratorTest.java

```diff
@@ -0,0 +1,92 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive;
+
+import java.util.concurrent.CompletionStage;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+import org.hibernate.cfg.Configuration;
+
+import org.junit.Test;
+
+import io.vertx.ext.unit.TestContext;
+
+public class IdentityGeneratorTest extends BaseReactiveTest {
+
+	// The number of entities we want to create
+	private static final int ENTITY_NUMBER = 100;
+
+	@Override
+	protected org.hibernate.cfg.Configuration constructConfiguration() {
+		Configuration configuration = super.constructConfiguration();
+		configuration.addAnnotatedClass( EntityWithIdentity.class );
+		return configuration;
+	}
+
+	private CompletionStage<?> populateDb() {
+		final Object[] identities = new EntityWithIdentity[ENTITY_NUMBER];
+		for ( int i = 0; i < ENTITY_NUMBER; i++ ) {
+			identities[i] = new EntityWithIdentity( i );
+		}
+		return getSessionFactory().withTransaction( (session, tx) -> session.persist( identities ) );
```

Comment:
@DavideD This test seems inadequate. We don't check that the generated identities are actually populated back into the entities. Remember that with autoincrement/identity columns, the whole lifecycle of the `persist()` operation is pretty different, so we need to properly test it.

Reply:
I see, ok

Reply:
Do you think this is enough? https://github.com/hibernate/hibernate-reactive/pull/290/commits/47863a684fae83686672710204fc138a3ef53c03

Reply:
Shouldn’t we be checking that the generated value is greater than 0?

Reply:
Done: https://github.com/hibernate/hibernate-reactive/pull/290/commits/edac32102af36b75bca3af61b281b08348558e3f#diff-bf65d08103f4e4a90f49edd42293fb94R43

Reply:
cool, thanks